### PR TITLE
Fix: validation of incoming payloads

### PR DESCRIPTION
**Pull request by kristjank**
## Proposed changes
Changed the validation of applyTransactions to take into account recent bug #925.
- Not allowing to to double votes (voting again when voted)
- Not allowing to vote for non-existing delegate (although) wallet exists
- In case of cold wallet, still check transaction conditions for vote...

## Types of changes
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist
- [x] I have read the [CONTRIBUTING](https://docs.ark.io/developers/guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by kristjank** [comment]
@faustbrian

File: packages/core-transaction-pool/lib/pool-wallet-manager.js [hunk 1]
```diff
@@ -94,17 +94,18 @@ module.exports = class PoolWalletManager extends WalletManager {
     if (!recipient && recipientId) { // cold wallet
       recipient = new Wallet(recipientId)
       this.walletsByAddress[recipientId] = recipient
+    }
```
Comment (faustbrian) [inline packages/core-transaction-pool/lib/pool-wallet-manager.js]: Did you make sure that breaking apart this `if else` block won't have any side effects?

Comment (kristjank) [reply packages/core-transaction-pool/lib/pool-wallet-manager.js]: sending data to cold wallet - so first if is true... so if the if-else remain together 

`  } else if (!sender.canApply(data)) {` is never executed and transaction is accepted with (0 fees for example or anything else.

-- 
we should then if - transaction is not applied/accepted - remove the cold wallet from wallet manager.. @fix

Comment (fix) [reply packages/core-transaction-pool/lib/pool-wallet-manager.js]: mmm true that. Can you document this  in the code or maybe extract these lines as a function:
https://github.com/ArkEcosystem/core/blob/fc4a29e866f03cd26ebc0a0b8ca1e0206becc4ee/packages/core-database/lib/wallet-manager.js#L212-L218

replaced by `const recipient = this.findOrCreateWallet(recipientId)`

Comment (faustbrian) [reply packages/core-transaction-pool/lib/pool-wallet-manager.js]: @fix https://github.com/ArkEcosystem/core/pull/933/commits/3126c3713d0025873f0fef45ca7c26ddb866cbdc

Should be ready to be merged.

**Pull request merged**